blocks/form/gdpr-confirmation.js

```diff
@@ -0,0 +1,90 @@
+function hasQueryParams() {
+  return !!window.location.search;
+}
+
+// Send a query parameters to the GDPR API Accept
+async function acceptResult(accept, urlProd, sucessRedirect, block) {
+  // get url parameters
+  const urlParams = new URLSearchParams(window.location.search);
+  const paramData = urlParams.get('data');
+  const paramInit = urlParams.get('init');
+  const url = urlProd + accept;
+  const options = {
+    method: 'PUT',
+    headers: { 'Content-type': 'application/json; charset=UTF-8' },
+    body: JSON.stringify({
+      data: paramData,
+      init: paramInit,
+      confirm: 'true',
+    }),
+  };
+  try {
+    const response = await fetch(url, options);
+    const resData = await response.text();
+    if (response.ok) {
+      window.open(sucessRedirect);
+    } else {
+      const { message } = JSON.parse(resData);
+      throw message;
+    }
+  } catch (error) {
+    const divError = document.createElement('div');
+    divError.className = 'diverror';
+    divError.innerHTML = 'Something went wrong. Try again';
+    block.appendChild(divError);
+    const span = document.createElement('span');
+    span.innerHTML = 'x';
+    span.className = 'closeerror';
+    divError.appendChild(span);
+    span.addEventListener('click', () => {
+      divError.style.display = 'none';
+    });
+  }
+}
+
+/* eslint-disable import/prefer-default-export */
+export default function decorate(block) {
+  const confirmationForm = document.createElement('form');
+  const configlIst = [];
+  const configlIstNames = [];
+  [...block.children].forEach((row) => {
```

Comment:
there is https://github.com/Netcentric/cn-website/blob/main/scripts/lib-franklin.js#L281 we can use for this